src/validr/_validator.pyx

```diff
@@ -115,6 +115,53 @@ def validator(bint string=False):
     return decorator
 
 
+_UNIQUE_CHECK_ERROR_MESSAGE = """
+Unable to check unique! only list of types below are available:
+    1. scalar type, eg: bool, int, float, str
+    2. list of scalar type
+    3. dict of scalar type value
```

Comment:
Don't really like this approach. That should be done based on some internal function (like .hash()). So, if user is up to adding this function to his own type, the unique check should be automatically possible. That's what i think a sane approach.

Reply:
Here is worth to improve.